Sources/PackageDescription/Product.swift

```diff
@@ -8,46 +8,125 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
-public struct Product {
-    public let name: String
-    public let type: ProductType
-    public let modules: [String]
+/// Defines a product in the package.
+public enum Product {
 
-    public init(name: String, type: ProductType, modules: String...) {
-        self.init(name: name, type: type, modules: modules)
+    /// An exectuable product.
+    public struct ExecutableProduct {
+
+        /// The name of the executable product.
+        public let name: String
+
+        /// The targets in the product.
+        public let targets: [String]
+    }
+
+    /// A library product.
+    public struct LibraryProduct {
+
+        /// The type of library products.
+        public enum LibraryType: String {
+            case `static`
+            case `dynamic`
+        }
+
+        /// The name of the library product.
+        public let name: String
+
+        /// The type of the library.
+        ///
+        /// If the type is unspecified, package manager will automatically choose an optimized type.
```

Comment:
I would leave out "optimized" here... we don't want to promise too much in the first version. :)  Later we can explain that the choice will be made based on how the product is used, but for now we can just say that the package manager will automatically choose what kind of library to build without going into details.